controllers/nodehealthcheck_controller_test.go

```diff
@@ -154,68 +143,75 @@ var _ = Describe("Node Health Check CR", func() {
 		})
 	})
 
+	createObjects := func(objects ...client.Object) {
+		for _, obj := range objects {
+			Expect(k8sClient.Create(context.Background(), obj)).To(Succeed())
+		}
+	}
+
+	deleteObjects := func(objects ...client.Object) {
+		for _, obj := range objects {
+			// ignore errors, CRs might be deleted by reconcile
+			_ = k8sClient.Delete(context.Background(), obj)
+		}
+	}
+
 	Context("Reconciliation", func() {
 		var (
-			underTest       *v1alpha1.NodeHealthCheck
-			objects         []runtime.Object
-			reconciler      NodeHealthCheckReconciler
-			upgradeChecker  fakeClusterUpgradeChecker
-			mhcChecker      mhc.DummyChecker
-			reconcileError  error
-			reconcileResult controllerruntime.Result
-			getNHCError     error
+			underTest *v1alpha1.NodeHealthCheck
+			objects   []client.Object
 		)
 
-		var setupObjects = func(unhealthy int, healthy int) {
+		setupObjects := func(unhealthy int, healthy int) {
 			objects = newNodes(unhealthy, healthy, false)
 			underTest = newNodeHealthCheck()
-			remediationTemplate := newRemediationTemplate()
-			objects = append(objects, underTest, remediationTemplate)
+			objects = append(objects, underTest)
 		}
 
 		JustBeforeEach(func() {
-			client := fake.NewClientBuilder().WithRuntimeObjects(objects...).Build()
-			reconciler = NodeHealthCheckReconciler{
-				Client:                      client,
-				Log:                         controllerruntime.Log.WithName("NHC Test Reconciler"),
-				Scheme:                      scheme.Scheme,
-				ClusterUpgradeStatusChecker: &upgradeChecker,
-				MHCChecker:                  mhcChecker,
-				Recorder:                    record.NewFakeRecorder(20),
+			createObjects(objects...)
+			// give the reconciler some time
+			time.Sleep(2 * time.Second)
```

Comment:
Can we use `Eventually` instead of flat timeout ?

Reply:
what should eventually test for?

Reply:
I've assumed you were using the timeout because the following statment failed without it (i.e `Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())` ), so I'd Expect `Eventually` to test that.

Reply:
No, I'm just giving the reconciler some time, and afterwards I get the latest version of the CR because it's needed in every test. I don't know what to expect in the CR though, it depends on the individual test case. It does not fail.